tableauserverclient/server/endpoint/jobs_endpoint.py

```diff
@@ -48,3 +54,35 @@ def get_by_id(self, job_id):
         server_response = self.get_request(url)
         new_job = JobItem.from_response(server_response.content, self.parent_srv.namespace)[0]
         return new_job
+
+    def wait_for_job(self, job_id, *, timeout=None):
+        id_ = getattr(job_id, "id", job_id)
+        wait_start_time = time.time()
+        logger.debug(f"Waiting for job {id_}")
+
+        current_sleep_interval = ASYNC_JOB_POLL_MIN_INTERVAL
+        job = self.get_by_id(id_)
+        while job.completed_at is None:
+            max_sleep_time = ASYNC_JOB_POLL_MAX_INTERVAL
+
+            if timeout is not None:
+                elapsed = (time.time() - wait_start_time)
+                if elapsed >= timeout:
+                    raise TimeoutError(f"Timeout after {elapsed} seconds waiting for asynchronous job: {id_}")
+                max_sleep_time = max(ASYNC_JOB_POLL_MIN_INTERVAL, timeout - elapsed)
+
+            time.sleep(min(current_sleep_interval, max_sleep_time))
+            job = self.get_by_id(id_)
+            current_sleep_interval *= ASYNC_JOB_POLL_BACKOFF_FACTOR
+            logger.debug(f"\tJob {id_} progress={job.progress}")
+
+        logger.info("Job {} Completed: Finish Code: {} - Notes:{}".format(id_, job.finish_code, job.notes))
+
+        if job.finish_code == 0:
```

Comment:
Either a quick comment or change the magic numbers to a constant so we remember for the future what the finish codes mean

Reply:
Added `JobItem.FinishCode` which has all finish codes as named constants